amethyst_rendy/src/bundle.rs

```diff
@@ -23,18 +23,64 @@ use amethyst_core::{
 use amethyst_error::{format_err, Error};
 use std::collections::HashMap;
 
+/// A bundle of systems used for rendering using `Rendy` render graph.
+///
+/// It's highly recommended to use `PluggableRenderingBundle` as it is easier to construct and manage
+/// at the loss of advanced manual handling.
+///
+/// This allows for fully customizable rendering processing by passing in your own `GraphCreator`.
+#[derive(Debug)]
+pub struct RenderingBundle<B, G>
+where
+    B: Backend,
+    G: GraphCreator<B>,
+{
+    graph_creator: G,
+    phantom_data: std::marker::PhantomData<B>,
+}
+
+impl<B: Backend, G: GraphCreator<B>> RenderingBundle<B, G> {
+    /// Creates a new `RenderingBundle` with a custom `GraphCreator`.
+    ///
+    /// It's highly recommended to use `PluggableRenderingBundle` as it is easier to construct and
+    /// manage at the loss of advanced manual handling.
+    ///
+    /// This allows for fully customizable rendering processing by passing in your own `GraphCreator`.
+    pub fn new(graph_creator: G) -> Self {
+        Self {
+            graph_creator,
+            phantom_data: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<'a, 'b, B: Backend, G: GraphCreator<B> + 'b> SystemBundle<'a, 'b> for RenderingBundle<B, G> {
+    fn build(self, builder: &mut DispatcherBuilder<'a, 'b>) -> Result<(), Error> {
+        builder.add(MeshProcessorSystem::<B>::default(), "mesh_processor", &[]);
+        builder.add(
+            TextureProcessorSystem::<B>::default(),
+            "texture_processor",
+            &[],
+        );
+        builder.add(Processor::<Material>::new(), "material_processor", &[]);
+
+        builder.add_thread_local(RenderingSystem::<B, G>::new(self.graph_creator));
+        Ok(())
```

Comment:
Did you forget to add the SpriteSheet processor before the RenderingSystem?
Also, it seems like you forgot to add the barrier.

Would it be possible to put those 4 systems (the 3 you defined + SpriteSheet) in a private/local small bundle called `RenderingProcessorBundle` or something similar?

Reply:
> Did you forget to add the SpriteSheet processor before the RenderingSystem?

Before, you had to declare the SpriteSheet processor explicitly; it was good because the 3D games doesn't need this system most of the times.

However on a second thought, following the above reasoning I should remove also the `MeshProcessorSystem`.
What do you think about this?

> Also, it seems like you forgot to add the barrier.

In the PluggableRenderingBundle there is a mechanism that allow the plugins to initialize some `System`s. These `System`s must always be executed after the one defined in the bundle.

Since this mechanism is not used with RenderingBundle, and it's meant to give 100% freedom to the developer is not necessary the barrier.

> Would it be possible to put those 4 systems (the 3 you defined + SpriteSheet) in a private/local small bundle called RenderingProcessorBundle or something similar?

Do you mean move `RenderingBundle` in another module and remove `builder.add_thread_local(RenderingSystem::<B, G>::new(self.graph_creator));` so it must be defined explicitly during the dispatcher creation?

Reply:
Solved by removing the `RenderingBundle` as discussed here: https://github.com/amethyst/amethyst/pull/1839#issuecomment-518305303